### DISCUSSION TITLE
check if inventory is from the player

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#1337] [Help] [answered]
Hello, I want to know if the inventory from the InventoryClickEvent or the InventoryPreClickEvent is from the player. I use this code to load the events: 
```
private val inventoryHandler = EventNode.value("invsee-inv-${player.uuid}", EventFilter.INVENTORY) { obj : Inventory -> obj == inventory || obj == target.inventory || obj == player.inventory }
```
but I got this error:
```
> java.lang.NullPointerException: Parameter specified as non-null is null: method nl.jochem.essentials.mechanics.InvseeHandler.inventoryHandler$lambda-1, parameter obj
``` 
How can I change my code to check if the inventory is from the player/target?

**Comment by GreatWyrm** [comment, marked as answer]
Check if the inventory in the event is null.

As specified in the [method javadocs](https://javadoc.minestom.net/net/minestom/server/event/inventory/InventoryPreClickEvent.html#getInventory()), if the getInventory() method returns null, that means a player inventory has been clicked.

**Reply by DevBlokJochem** [reply]
But how can I get the owner of the playerinventory if the inventory is null?

**Reply by GreatWyrm** [reply]
The getPlayer() method in the event. I don't think a situation will arise in which a player can open another player's inventory.

**Reply by DevBlokJochem** [reply]
True, but can I check it in the eventnode? If the player/target's inventory got clicked? So I can filter the other players out of the events.

**Reply by GreatWyrm** [reply]
Something like this, I think:
`.filter(inventoryPreClickEvent -> inventoryPreClickEvent.getPlayer() == player && inventoryPreClickEvent.getInventory() == null)`
